### PR TITLE
update the Service Endpoint section of the spec

### PR DESCRIPTION
**Pull request by csuwildcat**

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -133,12 +133,12 @@ The following DID Document Service Endpoint entries ****MUST**** be present in t
...
 ```json
 {
-  "id": "did:example:123",
+  "id": "did:example:alice",
   "service": [{
     "id":"#dwn",
     "type": "DecentralizedWebNode",
     "serviceEndpoint": {
-      "nodes": ["https://dwn.example.com", "https://example.org/dwn"]
+      "nodes": ["did:example:host", "https://dwn.example.com"]
```
Comment (andorsk) [inline spec/spec.md:141]: @csuwildcat looking at https://github.com/decentralized-identity/decentralized-web-node/blob/dc69c89ffb12eb304f796f368f65b78e172d8826/spec/spec.md?plain=1#L202, you mention it resolves to "https://dwn.example.com". Per this example, given the preference toward resolving toward the first listed node in the service endpoint section, wouldn't it resolve to did:example:host first, which may or may not resolve eventually to `https://dwn.example.com`? Wouldn't it make more sense to list `https://dwn.example.com` first, so the reference below is more consistent?

File: spec/spec.md [hunk 2]
```diff
@@ -186,30 +186,34 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
...
 
 ```json
-did:example:123?service=DecentralizedWebNode&queries=W3sgTUVTU0FHRV8xIH0sIHsgTUVTU0FHRV9OIH1d...
+did:example:alice?service=DecentralizedWebNode&message=W3sgTUVTU0FHRV8xIH0sIHsgTUVTU0FHRV9OIH1d...
 ```
 
 **Resolve DID to locate its Decentralized Web Node URIs:**
 
-`did:example:123` -->  resolve to Decentralized Web Node endpoint(s)  -->  `https://dwn.example.com/`
+`did:example:alice --> resolve DWN endpoint(s) --> `https://dwn.example.com`
```
Comment (andorsk) [inline spec/spec.md:202]: See my point above. We should change the example to have `https://dwn.example.com` as the first listed endpoint, given the service resolution preferences described in https://identity.foundation/decentralized-web-node/spec/#resolution, #3.

**Review by andorsk — COMMENTED** [review]
@csuwildcat thanks for putting this together. Because of the preferences described in https://identity.foundation/decentralized-web-node/spec/#resolution, it is a little confusing when you swap out to resolve against the second item: "https://example.org/dwn" instead of did:example:host. Maybe we can make a note somewhere that we are resolving to the second item for explanatory purposes?

**Review by andorsk — COMMENTED** [review]
@csuwildcat thanks for putting this together. Because of the preferences described in https://identity.foundation/decentralized-web-node/spec/#resolution, it is a little confusing when you swap out to resolve against the second item: "https://example.org/dwn" instead of did:example:host. Maybe we can make a note somewhere that we are resolving to the second item for explanatory purposes?

File: spec/spec.md [hunk 3]
```diff
@@ -186,30 +186,34 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
 **DID-Relative URL example for passing multiple messages:**
 
 ::: note
```
Comment (andorsk) [inline spec/spec.md:188]: Could you clarify here how this "DID-Relative URL example for passing multiple messages". I don't see multiple messages here. Am I missing something?

File: spec/spec.md [hunk 4]
```diff
@@ -186,30 +186,34 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
...
 
 **Resolve DID to locate its Decentralized Web Node URIs:**
 
-`did:example:123` -->  resolve to Decentralized Web Node endpoint(s)  -->  `https://dwn.example.com/`
+`did:example:alice --> resolve DWN endpoint(s) --> `https://dwn.example.com`
+
+`did:example:alice`--> resolve DWN endpoint(s) --> `did:example:host` --> resolve DWN endpoint(s) --> `https://dwn.did-relative-host.com`
+
+::: note
+Only DID URIs in the `nodes` array will be allowed to initiate sync activities with other DWeb Node instances.
```
Comment (andorsk) [inline spec/spec.md:207]:
```suggestion
Only DID URIs in the `nodes` array will be allowed to initiate [sync](https://identity.foundation/decentralized-web-node/spec/#sync) activities with other DWeb Node instances.
```

File: spec/spec.md [hunk 5]
```diff
@@ -186,30 +186,34 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
 **DID-Relative URL example for passing multiple messages:**
 
 ::: note
-For example purposes, the `queries` parameter value below is neither JSON stringified nor Base64Url encoded, but should be when using Decentralized Web Node URLs in practice (see the [DID-relative URL Composition](#composition) instructions above).
+For example purposes, the `message` parameter value below is neither JSON stringified nor Base64Url encoded, but should be when using Decentralized Web Node URLs in practice (see the [DID-relative URL Composition](#composition) instructions above).
```
Comment (TallTed) [inline spec/spec.md:189]:
```suggestion
***Note:** For example purposes, the `message` parameter value below is neither JSON stringified nor Base64Url encoded, but should be when using Decentralized Web Node URLs in practice (see the [DID-relative URL Composition](#composition) instructions above).*
```

**Review by TallTed — CHANGES_REQUESTED** [review]
A little formatting to increase comprehension

**Comment by peacekeeper** [comment]
I think this looks good, I have always liked how you can pass a message as part of a DID URL.

File: spec/spec.md (hunk 5 above)
Comment (andorsk) [reply spec/spec.md:189]: Thank you @TallTed

**Comment by andorsk** [comment]
@csuwildcat any thoughts or comments on the reviews? Would love to get this in asap.

**Comment by csuwildcat** [comment]
Left a comment

**Comment by LiranCohen** [comment]
Working with the Web5 library, I noticed some additional keys within the Service Endpoint for DWNs.

```
  messageAttestationKeys?: string[]
  messageAuthorizationKeys?: string[] 
  nodes: string[]
  recordEncryptionKeys?: string[] 
```

Are these definitively needed here, or they a temporary placeholder? Would like to get the spec updated to match and describe the fields if so.

File: spec/spec.md [hunk 6]
```diff
@@ -186,7 +186,7 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
 **DID-Relative URL example for passing multiple messages:**
```
Comment (TallTed) [inline spec/spec.md:186]:
```suggestion
**DID-Relative URL example for passing messages:**
```

File: spec/spec.md (hunk 3 above)
Comment (TallTed) [reply spec/spec.md:188]: Suggested fix, [above](https://github.com/decentralized-identity/decentralized-web-node/pull/228/files/dc69c89ffb12eb304f796f368f65b78e172d8826..a10509fab17347aa8db22cf39329fbf3670a9fcf#r1220696488).

File: spec/spec.md [hunk 7]
```diff
@@ -186,7 +186,7 @@ The following process defines how a DID-Relative URL for a Decentralized Web Nod
...
 
 ::: note
-For example purposes, the `message` parameter value below is neither JSON stringified nor Base64Url encoded, but should be when using Decentralized Web Node URLs in practice (see the [DID-relative URL Composition](#composition) instructions above).
+***Note:** For example purposes, the `message` parameter value below is neither JSON stringified nor Base64Url encoded, but should be when using Decentralized Web Node URLs in practice (see the [DID-relative URL Composition](#composition) instructions above).*
 :::
```
Comment (TallTed) [inline spec/spec.md:190]:
```suggestion
```

Comment (csuwildcat) [reply spec/spec.md:190]: This already renders as a note in UI, so there's no real need to again say Note within the note.

**Pull request merged**